configure.ac

```diff
@@ -488,10 +488,17 @@ AM_CONDITIONAL(SYSD_SYSUSERS, test "x$systemd_sysusers" = "xyes")
 AC_CHECK_PROG(systemd_tmpfiles, systemd-tmpfiles, yes)
 AM_CONDITIONAL(SYSD_TMPFILES, test "x$systemd_tmpfiles" = "xyes")
 # Check all tools used by make install
+
+AC_CHECK_PROG(busybox,busybox,yes)
+AM_CONDITIONAL(BUSYBOX, [test x"$busybox" = xyes])
```

Comment:
So on a lot of devices, busybox is sometimes renamed. I think this would be more generic if we look for adduser or addgroup during configure time over checking for busybox. What about in a side-by-size install where they have busybox and groupadd? We could still use groupadd.

Reply:
I tried to install groupadd in Alpine but it was not available.  If the name of busybox is often changed we should determine the name of the running shell and compare it with readlink from the path of adduser.

Reply:
It's possible to determine the shell on Linux via `/proc/$$/exe`. But perhaps it would be better to use the current method from this PR for systems which do not change the name (works for Alpine) and add a configure option --with-busybox for cases when the name was changed?

Reply:
We don't care what shell we're in, so long as its compatible, and we don't care about busybox. We care that we have adduser or useradd and the same for group. We just need one available in each set and you can run. So lets not worry about looking at our shell or looking for busybox, rather lets look for something executable that has what we need. I checked locally to make sure that the autoconf macros pick up symlinked executables, since that's how busybox is installed, and it does seem to work as expected.
```
AC_CHECK_PROG(addgroup, ...)
AC_CHECK_PROG(groupadd, ...)
AC_CHECK_PROG(adduser, ...)
AC_CHECK_PROG(useradd, ...)

if not useradd and not adduser:
   fail("Need adduser or useradd)

if not groupadd and not addgroup:
   fail("Need addgroup or groupadd)
```

Reply:
The busybox adduser and e.g. ubuntu adduser have different options. So would you expect?:
```
if not groupadd and adduser:
   use busybox version of adduser for make installl
elsif groupadd
   use groupadd for make install
else 
   fail("Need addgroup or groupadd")
```